src/compression/lz4.rs

```diff
@@ -0,0 +1,502 @@
+use std::mem;
```

Comment:
Whoa, this is a lot of low-level code. Did you write this all yourself? If not, where did it come from? Do we need to have all this code, or is there a crate we can use?